responsibleai/responsibleai/_managers/causal_manager.py

```diff
@@ -234,6 +231,27 @@ def add(
         self._results.append(result)
         return result
 
+    def _validate_train_test_categories(
+        self,
+        X_train: pd.DataFrame,
+        X_test: pd.DataFrame,
+        categoricals,
+    ):
+        discovered = {}
+        for column in X_train.columns:
+            if column in categoricals:
+                train_unique = np.unique(X_train[column])
+                test_unique = np.unique(X_test[column])
+                difference = np.setdiff1d(test_unique, train_unique)
+                if difference.shape[0] != 0:
+                    discovered[column] = difference.tolist()
+        if len(discovered) > 0:
+            message = ("Causal analysis requires that every category of "
+                       "categorical features present in the test data "
+                       "be also present in the train data. "
+                       "Categories missing from train data: {}")
+            raise UserConfigValidationException(message.format(discovered))
```

Comment:
Should this function be called in the init time of model analysis class. Looks like a generic validation that canbenefit all components.

Reply:
If my understanding is correct, explanations handle this case by dropping categories missing in the training data, so by moving this to the top level we would fail explanations with a user config error unnecessarily.

Whether counterfactuals and error analysis handle this or not is a valid question that I think @xuke444 already validated and found that they both can handle this case.

Reply:
So I think we want to keep this validation causal-specific, but I'm open to moving it if any of my assumptions above are wrong!

Reply:
I don't think this is specific to causal per say. Sure CFE and EA don't result in errors but that doesn't indicate whether these components handle this situation correctly.  To me this feels like generic enough and a good case for users to know that something is amiss in their validation data.

Reply:
Having categories (or classes) in the validation/test data unseen in the training data is a very common scenario that is often handled by simply dropping unknown categories. sklearn has this built in: https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html

I'm not sure I understand the concern regarding "whether these components handle this situation correctly"

Reply:
Tagging @imatiach-msft since I know he is very familiar with ignoring unknown categoricals from work on interpret transformations

Reply:
discussed on teams, seems fine to me

Reply:
I will check this PR in to unblock causal analysis scenarios where train and test data need to be disjoint sets, however we can add this same validation for other components in a future PR if we can agree that it is required.